python/ray/air/tests/test_checkpoints.py

```diff
@@ -480,6 +482,42 @@ def test_dict_cp_as_directory(self):
 
         assert not os.path.exists(checkpoint_dir)
 
+    def test_fs_cp_from_populator(self):
+        # Create checkpoint from populator
+        checkpoint = Checkpoint.from_directory_populator(
+            lambda dir: (Path(dir) / "data.txt").write_text("Data")
+        )
+
+        # Ephemeral checkpoints shouldn't have their URI set
+        assert not checkpoint.uri
+
+        # Assert data is correctly saved
+        with checkpoint.as_directory() as dir:
+            assert (Path(dir) / "data.txt").read_text() == "Data"
+
+        # Simulate pickling (e.g. sending via ray object store)
+        cp_blob = cloudpickle.dumps(checkpoint)
+
+        # Assert that local data is deleted when checkpoint is deleted
+        checkpoint_path = checkpoint._local_path
+        del checkpoint
+        assert not Path(checkpoint_path).exists()
+
+        # Restore checkpoint from pickled data
+        checkpoint = Checkpoint.from_directory(
+            cloudpickle.loads(cp_blob).to_directory()
+        )
+
+        # Assert content is the same
+        checkpoint_path = checkpoint._local_path
+        with checkpoint.as_directory() as dir:
+            assert (Path(dir) / "data.txt").read_text() == "Data"
+
+        # Assert that this checkpoint is not managed anymore, so that data remains
+        # on disk.
+        del checkpoint
```

Comment:
When will this be called and by whom? How do we enforce that this is "eventually" called?
For example, in the case of lazy checkpoint.